utils/near-framed-read/src/framed_read.rs

```diff
@@ -0,0 +1,296 @@
+use std::borrow::BorrowMut;
+use std::io;
+use std::pin::Pin;
+use std::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use bytes::BytesMut;
+use futures::channel::mpsc::{UnboundedReceiver, UnboundedSender};
+use futures::SinkExt;
```

Comment:
Ok, I think this is actually the core problem of why `remove_msg` creates problems. 

The TL;DR here is that we shouldn't be using `futures` crate, and instead use channels from tokio. They have an explicit, documented solution for bridging sync and async worlds: 

https://docs.rs/tokio/1.13.0/tokio/sync/mpsc/index.html#communicating-between-sync-and-async-code

More generally, we shouldn't be using the `futures` crate anywhere, and instead stick to pure `tokio`. Some historical context: 

The `futures` crate was was historically the first approach to non-blocking Rust, designed well in advance of `async/await` language support. `async/await` was retrofitted onto the `futures`, but not perfectly -- by that time, `futures` was already a widely used crate, so it had a lot of constraints on API evolution. 

After async-await shipped, community spend a lot of iteration on designing tokio and alternative runtimes (smol, async-std). At that time, the advice was "try to stick to futures crate, as that is runtime agnostic". 

Today, tokio actually shipped a stable 1.0 version. Now, it esstially fulfils async-std vision -- an almost drop-in replacement for types from std, compatible with epoll event loop. Everything that the `futures`  crate used to do is now available in tokio, and generally has better design. Channels are just an instance of this: tokio version takes lessons learned from futures, and makes "send from blocking to async world" an explicitly supported pattern.

Reply:
@matklad Thanks, thanks for the update. Yes, in the past using `futures` crate was the recommended solution.

Anyway, I stopped using the `futures` crate in `framed_read.rs`. And I switched to using `tokio::sync::mpsc::unbounded_channel::<()>();` instead.
Please take a look.